docs/dev-client-networking.md

```diff
@@ -3,259 +3,647 @@ id: dev-client-networking
+
+- A `deleted` entity was explicitly deleted by a client. That is, someone
+pressed a button or took some action to delete it on purpose. Entities that
+have been `deleted` cannot be recreated.
+- A `removed` entity was removed incidentally. For example, it may have been
+removed when the `creator` disconnected from the room. If the `creator`
+reconnects and sends a `CreateMessage` with a matching `networkId`, it is
+acceptable to recreate the entity.
+
+
+#### ClientJoin
+
+These tell a client that a new client has connected. The next time the
+`networkSendSystem` runs, the receiving client will send the new client
+messages about entities it is the `creator` of, and update messages for
+entities it is the `owner` of.
+
+
+#### ClientLeave
+
+These tell a client that another client has disconnected. The next time the
+`networkReceiveSystem` runs, the receiving client will `remove` entities that
+the disconnected client was the `creator` of.
+
+
+#### CreatorChange
+
+These tell a client that the `creator` of an entity has been reassigned.
+Typically, this means that an entity has been `pinned` (or `unpinned`), and
+reticulum has assigned (or unassigned) itself as the entity&rsquo;s `creator`.
+
+
+### Ownership
+
+Most of the complexity in the `networkSendSystem` and the
+`networkReceiveSystem` stem from this property of the network. Here are some
+examples where this complexity reveals itself:
+
+- The `lastOwnerTime` is used to ensure that ownership transfer is handled
+identically by all clients, even when messages arrive out of order.
+- The `deletedNids` collection ensures that out-of-order `CreateMessage`s do
+not cause `deleted` entities to be accidentally recreated.
+- The `storedUpdates` allows a client to save `UpdateMessage` s it has received
+but has no way to process, as can happen when it receives an `UpdateMessage`
+from the `owner` of an entity before it receives a `CreateMessage` from its
+`creator`.
+- The `takeSoftOwnership` function allows clients to take ownership of unowned
+entities in such a way that only clients with the most recent information about
+that entity will be eligible as the new owner.
+
+For the most part, users of the networking systems do not need to understand
+these concepts. These are handled internally by the systems themselves. However,
+users do need to understand that ownership is not transactional or guaranteed.
+That is, ownership is not &ldquo;requested and then transferred&rdquo;, and
+just because one client claims ownership of an entity does not mean that other
+clients will respect that claim.
+
+Users can inspect the state the `Networked` or `Owned` components as needed in
+cases when their ownership claims matter. They may find themselves writing
+coroutines that looks like this:
+
+```typescript
+takeSoftOwnership(world, eid);
+yield sleep( 3000 ); // Wait a few seconds to see if we "win" ownership
+if (!hasComponent(world, Owned, eid)) return;
+```

```

Comment:
This code example opens up more a lot of questions. What kinds of ownership actions would make sense with a 3 second latency? When should I use this pattern rather than takeOwnership() ?

Reply:
Same above.

I hope we write the details about the "Ownership" concept in the first "Ownership" section in the follow up PRs.

Reply:
Added a TODO in #160 for now, thanks.